core/instance_resource_crud.go

```diff
@@ -74,8 +82,75 @@ func (s *InstanceResourceCrud) Create() (e error) {
 	return
 }
 
+/*
+ * Return the id of the first VNIC attached to this Instance.
+ *
+ * NOTE while the instance is still being created, calls to this function
+ * can return  an error priort to the Vnic being attached.
+ */
+func (s *InstanceResourceCrud) getInstanceVnicId() (vnic_id string, e error) {
+	compartmentID := s.Resource.CompartmentID
+
+	opts := &baremetal.ListVnicAttachmentsOptions{}
+	options.SetListOptions(s.D, &opts.ListOptions)
+	opts.AvailabilityDomain = s.Resource.AvailabilityDomain
+	opts.InstanceID = s.Resource.ID
+
+	var list *baremetal.ListVnicAttachments
+	if list, e = s.Client.ListVnicAttachments(compartmentID, opts); e != nil {
+		return "", e
+	}
+
+	if len(list.Attachments) < 1 {
+		log.Printf("[DEBUG] GetInstanceVnicID - InstanceID: %q, State: %q, no vnic attachments: %q", s.Resource.ID, s.Resource.State, e)
+		return "", e
+	}
+
+	return list.Attachments[0].VnicID, nil
```

Comment:
@codycushing At launch we can only ever have a single vNIC, the instance has to be available before MvNICs can be attached. I don't think MvNIC is going to have any impact here.